modules/mod-deep-learning/ModelCard.cpp

```diff
@@ -0,0 +1,371 @@
+   {
+      // Input JSON is invalid according to the schema
+      // Output diagnostic information
+      TranslatableString msg = XO(
+         "A schema violation was found in the Model Card.\n "
+         "- violation found in URI: %s\n"
+         "- schema field violated: %s\n"
+         "- invalid document: %s\n"
+         "- schema document: %s\n"
+      );
+
+      rapidjson::StringBuffer sb;
+
+      validator.GetInvalidSchemaPointer().StringifyUriFragment(sb);
+      wxString uri(sb.GetString());
+      sb.Clear();
+
+      wxString invalidSchemaField(validator.GetInvalidSchemaKeyword());
+
+      rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+      doc.Accept(writer);
+      wxString invalidDocument(sb.GetString());
+      sb.Clear();
+   
+      rapidjson::Writer<rapidjson::StringBuffer> swriter(sb);
+      schema.Accept(swriter);
+      wxString schemaDocument(sb.GetString());
+
+      throw InvalidModelCardDocument(msg.Format(uri, invalidSchemaField, 
+                                                invalidDocument, schemaDocument), "");
+   }
+}
+
+void ModelCard::SerializeToFile(const std::string &path) const
+{
+   rapidjson::StringBuffer sb;
+   rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+
+   writer.StartObject();
+   Serialize(writer);
+   writer.EndObject();
+
+   wxFile file(path, wxFile::write);
+   if (!file.Write(wxString(sb.GetString())))
+      throw InvalidModelCardDocument(XO("Could not serialize ModelCard to file"), "");
+}
+
+void ModelCard::DeserializeFromFile(const std::string &path, const Doc& schema)
+{
+   Doc d = parsers::ParseFile(path);
+   Deserialize(d, schema);
+}
+
+template < typename Writer >
+void ModelCard::Serialize(Writer &writer) const
+{
+   // name
+   writer.String("name");
+   writer.String(m_name.c_str());
+
+   // author
+   writer.String("author");
+   writer.String(m_author.c_str());
+   
+   // long description
+   writer.String("long_description");
+   writer.String(m_long_description.c_str());
+
+   // short description
+   writer.String("short_description");
+   writer.String(m_short_description.c_str());
+
+   // sample rate
+   writer.String("sample_rate");
+   writer.Int(m_sample_rate);
+
+   // multichannel
+   writer.String("multichannel");
+   writer.Bool(m_multichannel);
+
+   // effect type
+   writer.String("effect_type");
+   writer.String(m_effect_type.c_str());
+
+   // domain tags
+   writer.String("domain_tags");
+   writer.StartArray();
+   for (auto tag : m_domain_tags)
+      writer.String(tag.c_str());
+   writer.EndArray();
+
+   // tags
+   writer.String("tags");
+   writer.StartArray();
+   for (auto tag : m_tags)
+      writer.String(tag.c_str());
+   writer.EndArray();
+
+   // labels
+   writer.String("labels");
+   writer.StartArray();
+   for(auto label : m_labels)
+      writer.String(label.c_str());
+   writer.EndArray();
+
+   // model size
+   writer.String("model_size");
+   writer.Uint64((uint64_t)m_model_size);
```

Comment:
I see `m_is_local` and `m_local_path` are omitted.  Should I think it's a possible bug or should a comment tell me that is intended?

Reply:
8d42c80dded1ba5dd6b0d02155d8f145c67549d2

Reply:
added a clarification

```
    // note: the fields "is_local" and "local_path"
   // are not deseralized because they are filled out
   // by the ModelManager when the ModelCard is saved to / loaded from disk. 
```

Reply:
8d42c80dded1ba5dd6b0d02155d8f145c67549d2

Reply:
8d42c80dded1ba5dd6b0d02155d8f145c67549d2